easybuild/framework/easyconfig/easyconfig.py

```diff
@@ -491,10 +491,31 @@ def dump(self, fp):
 
         last_keys = ['sanity_check_paths', 'moduleclass']
 
+        tmp = self.enable_templating
+        self.enable_templating = False
+
         # build dict of default values
         default_values = dict([(key, DEFAULT_CONFIG[key][0]) for key in DEFAULT_CONFIG])
         default_values.update(dict([(key, self.extra_options[key][0]) for key in self.extra_options]))
 
+        template_values = dict([(const[1], const[0]) for const in TEMPLATE_CONSTANTS])
+
+        def replace_templates(value):
```

Comment:
but this right next to `resolve_templates` as a top-level function, rather than making it an inner function (inner functions should only be used by exception)